lxd-migrate/main_migrate.go

```diff
@@ -261,11 +264,21 @@ func (c *cmdMigrate) runInteractive(server lxd.InstanceServer) (cmdMigrateData,
 
 	config.InstanceArgs = api.InstancesPost{
 		Source: api.InstanceSource{
-			Type: "migration",
-			Mode: "push",
+			Type:              "conversion",
+			Mode:              "push",
+			ConversionOptions: c.flagConversionOpts,
 		},
 	}
 
+	// If server does not support conversion, fallback to migration.
+	// Migration will move the image to the server and import it as
+	// LXD instance. This means that images of different formats,
+	// such as VMDK and QCow2, will not work.
+	if !server.HasExtension("instance_import_conversion") {
```

Comment:
If the --format flag has been provided then we should fail in this case?

Reply:
We do not have format flag that would hint the image type. And the conversion option `format` is enabled by default.

However, since we are already using command `file` to detect whether the image is in raw format (to skip unnecessary `format`ting) we can do similar to prevent image upload if image type is not raw. - Still the user needs to go through the questions to tell us the source of image.